workflow/controller/operator.go

```diff
@@ -1485,9 +1513,9 @@ func (n loopNodes) Swap(i, j int) {
 
 // processAggregateNodeOutputs adds the aggregated outputs of a withItems/withParam template as a
 // parameter in the form of a JSON list
-func (woc *wfOperationCtx) processAggregateNodeOutputs(templateName string, scope *wfScope, prefix string, childNodes []wfv1.NodeStatus) {
+func (woc *wfOperationCtx) processAggregateNodeOutputs(tmpl *wfv1.Template, scope *wfScope, prefix string, childNodes []wfv1.NodeStatus) error {
```

Comment:
I added the return error value because new logic there needed at the time of the implementation but changed the logic later but forgot to remove the return value. I'll remove it.

Reply:
However, the original code ignores an error from `Marshal` which is not good. I'll keep the return value but change it to handle the error.